pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap

```diff
@@ -959,6 +959,50 @@ g {
 }
 nodes_that_need_write = [75, 74, 73, 66, 65, 64, 63, 62, 61, 54, 53, 52, 45, 44, 43, 33, 32, 31, 27, 26, 25, 15, 14, 13, 6, 5, 4, 0]
 
+g {
+	n[0]: &_43 _ => _44 @ bb32[9]: fn main_0; _44 = &raw mut _43;
+}
```

Comment:
Since `_44` is address-taken, this graph should look like `AddrOfLocal(_43) -> StoreValue`, and there should be a corresponding `StoreAddr` in the graph below that starts with `AddrOfLocal(_44)`.

The `p = *_4; *p = 1` code should also produce a `LoadValue -> p -> StoreAddr` chain somewhere, possibly in this graph (since `p` will concretely equal `&_43`).

Reply:
Which statement would the `StoreValue` result from? From my understanding it would need a store on the LHS, and a RHS that evaluates to the value `_43`, but I don't see where that is. My guess is that it must be `*p = 1`?

Reply:
It should come from the statement `_44 = &raw mut _43`.  Since `_44` is address-taken, we treat this like

```Rust
a = &mut _44;
*a = &raw mut _43;
```

Which produces the chains `AddrOfLocal(_44) -> a -> StoreAddr` and `AddrOfLocal(_43) -> StoreValue`.

Reply:
That makes sense, thank you